File: 04-estructuras-datos/13-seleccion-y-combinacion-estructuras/scala/practicas/GestionInscripciones.scala
case class Inscripcion(est:String,curso:String)
object GestionInscripciones:
  def main(args:Array[String]):Unit =
    val orden=Vector(Inscripcion("U1","S1")); val porEst=orden.groupBy(_.est); val cursos=orden.map(_.curso).toSet; println((orden,porEst,cursos))

File: 04-estructuras-datos/13-seleccion-y-combinacion-estructuras/java/practicas/GestionInscripciones.java
import java.util.*;

public class GestionInscripciones {
    record Inscripcion(String estudiante, String curso) {}
    public static void main(String[] args) {
        List<Inscripcion> orden = new ArrayList<>();
        Map<String, List<Inscripcion>> porEstudiante = new HashMap<>();
        Set<String> cursos = new HashSet<>();

        Inscripcion i = new Inscripcion("U001", "JAVA-01");
        orden.add(i);
        porEstudiante.computeIfAbsent(i.estudiante(), k -> new ArrayList<>()).add(i);
        cursos.add(i.curso());

        // La combinación evita exigir a una sola estructura operaciones incompatibles entre sí.
        System.out.println(orden);
        System.out.println(porEstudiante.get("U001"));
        System.out.println(cursos);
    }
}
